2014_TPDS_basic_decomposition_based_algorithm/src/handle_segment.rs

```diff
@@ -0,0 +1,155 @@
+use lib::graph_extension::{GraphExtension, NodeData};
+use petgraph::Graph;
+
+#[allow(dead_code)] //TODO: remove
+pub enum SegmentClassification {
+    Heavy,
+    Light,
+}
+
+pub struct Segment {
+    pub nodes: Vec<NodeData>,
+    pub begin_range: i32,
+    pub end_range: i32,
+    pub deadline: f32,                         //TODO: use
+    pub classification: SegmentClassification, //TODO: use
+}
+
+#[allow(dead_code)] //TODO: remove
+pub fn create_segments(dag: &mut Graph<NodeData, i32>) -> Vec<Segment> {
+    dag.calculate_earliest_finish_times();
+
+    let mut earliest_finish_times = Vec::new();
+
+    for node in dag.node_weights_mut() {
+        earliest_finish_times.push(node.params["earliest_finish_time"]);
+    }
+
+    earliest_finish_times.sort();
+    earliest_finish_times.dedup();
```

Comment:
```suggestion
    earliest_finish_times.dedup();
    earliest_finish_times.sort();
```

Reply:
Certainly this one is faster to sort.
Fixed.